src/ng/parse.js

```diff
@@ -1788,6 +1797,7 @@ function $ParseProvider() {
             }
             cache[cacheKey] = parsedExpression;
           }
+          parsedExpression = expensiveChecksInterceptor(parsedExpression, expensiveChecks);
```

Comment:
Also, this would hide properties like `assign`, `constant`, `inputs` and `literal`.